### PR TITLE
JPA spi example added

### PR DESCRIPTION
**Pull request by psytester**
Added: how to configure spi for JPA migration settings

Closes discussions #11099 as found solution in issue #10871

### CODE REVIEW

**Review by stianst — CHANGES_REQUESTED** [review]
We should rather document this in the configuring the database guide. I thought it was already covered somewhere, but probably mistaken

**Comment by psytester** [comment]
@stianst what do you expect here? Should I rebuild the PR and add the extension into database guide or do you wait for a review of @DGuhr / @pedroigor?

**Comment by stianst** [comment]
@psytester if you update the PR to add a section to https://github.com/keycloak/keycloak/blob/main/docs/guides/src/main/server/db.adoc instead of the provider config guide that'd be great.

**Comment by pedroigor** [comment]
@psytester Thanks. Can you please update the database guide as suggested by Stian?

I'm also wondering if it makes sense to provide some configuration options instead of using the SPI configuration? Not sure because of the upcoming changes to storage though.

**Review by DGuhr — CHANGES_REQUESTED** [review]
- I also think the right place would be a `setting the migration strategy for JPA` section in the db guide instead. 
- I also think it makes sense to have the config exposed in a more convenient way, e.g. `db-migration-strategy`, but i would suggest to handle this as a follow up. Created #11678 to discuss this further

**Comment by psytester** [comment]
Change is done.

Which magic does add the chapter "Relevant options" to the page to show config parameter and they possible values in a table? https://www.keycloak.org/server/db

**Review by pedroigor — APPROVED** [review]

**Review by pedroigor — DISMISSED** [review]

**Comment by DGuhr** [comment]
@psytester Sorry for reaching back late. The magic is in the first lines: 

```
... includedOptions="db db-* transaction-xa-enabled"
```

but this magic does not work for SPI-settings so-far, as they are not wrapped in the new configuration layer (it only works for configuration we have a [mapper](https://github.com/keycloak/keycloak/tree/main/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers) for, and as the SPIs are loaded differently (not "wrapped" by this layer), they can't be rendered in the section atm. That said, there's https://github.com/keycloak/keycloak/issues/11678 to change this, but I have not gotten an answer there so-far from the store team and at the moment we are restructuring the config into its own package in https://github.com/keycloak/keycloak/pull/12435 so I hesitate to change sth before this is done. 

@pedroigor this opens up a kind-of interesting question, that is, can and should we include the spi options into the docs "relevant options" table another way?

Apart from that i'll also approve here for now, as it should help ppl nevertheless :) thanks.

**Review by DGuhr — DISMISSED** [review]

File: docs/guides/src/main/server/configuration-provider.adoc [hunk 1]
```diff
@@ -42,6 +42,8 @@ Provider configuration options are provided when starting the server, as shown i
 .Setting the `connection-pool-size` for the `default` provider of the `connections-http-client` SPI
 <@kc.start parameters="--spi-connections-http-client-default-connection-pool-size=10"/>
 
+As usual, the configuration options can also be set in `conf/keycloak.conf` with follow-up start `build` option.
+
```
Comment (stianst) [inline docs/guides/src/main/server/configuration-provider.adoc:46]: Please remove as this is an irrelevant change

```suggestion
```

**Review by stianst — APPROVED** [review]

**Comment by pedroigor** [comment]
@stianst The example is now wrong after the new store integration. The options should use `legacy` instead of `quarkus` as the provider ir.

**Comment by psytester** [comment]
> use `legacy` instead of `quarkus` as the provider.

Was my first thought too

**Comment by pedroigor** [comment]
@psytester https://github.com/keycloak/keycloak/pull/14038

**Pull request merged**